airflow/settings.py

```diff
@@ -77,7 +80,7 @@
 DAGS_FOLDER: str = os.path.expanduser(conf.get('core', 'DAGS_FOLDER'))
 
 engine: Optional[Engine] = None
-Session: Optional[SASession] = None
+Session: Callable[..., SASession]
```

Comment:
This will conflict with https://github.com/apache/airflow/pull/20000 (and also , the SessionMaker is both callable, _and_ behaves like an actual Session. SQLA is weird like that)

Reply:
This might be fine though

Reply:
Let me see if I can just make this `Session: SessionMaker`

Reply:
Hmm no, SQLAlchemy is not type hinted well enough, annotating this as `sessionmaker` would make `Session()` return `Any`. I’ll keep this for now since this is more useful in general.

Reply:
Oh actually :)

Reply:
This is probably fine as it is (and it's how we use it in Airflow)